internal/connector/pingone_platform/resources/pingone_gateway_role_assignment.go

```diff
@@ -0,0 +1,82 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/patrickcping/pingone-go-sdk-v2/management"
+	"github.com/pingidentity/pingctl/internal/connector"
+	"github.com/pingidentity/pingctl/internal/logger"
+)
+
+// Verify that the resource satisfies the exportable resource interface
+var (
+	_ connector.ExportableResource = &PingoneGatewayRoleAssignmentResource{}
+)
+
+type PingoneGatewayRoleAssignmentResource struct {
+	clientInfo *connector.SDKClientInfo
+}
+
+// Utility method for creating a PingoneGatewayRoleAssignmentResource
+func GatewayRoleAssignment(clientInfo *connector.SDKClientInfo) *PingoneGatewayRoleAssignmentResource {
+	return &PingoneGatewayRoleAssignmentResource{
+		clientInfo: clientInfo,
+	}
+}
+
+func (r *PingoneGatewayRoleAssignmentResource) ExportAll() (*[]connector.ImportBlock, error) {
+	l := logger.Get()
+
+	l.Debug().Msgf("Fetching all %s resources...", r.ResourceType())
+
+	apiExecuteFunc := r.clientInfo.ApiClient.ManagementAPIClient.GatewaysApi.ReadAllGateways(r.clientInfo.Context, r.clientInfo.ExportEnvironmentID).Execute
+	apiFunctionName := "ReadAllGateways"
+
+	gatewaysEmbedded, err := GetManagementEmbedded(apiExecuteFunc, apiFunctionName, r.ResourceType())
+	if err != nil {
+		return nil, err
+	}
+
+	importBlocks := []connector.ImportBlock{}
+
+	l.Debug().Msgf("Generating Import Blocks for all %s resources...", r.ResourceType())
+
+	for _, gatewayInner := range gatewaysEmbedded.GetGateways() {
+		// Only PingFederate Connections have role assignments
+		if gatewayInner.Gateway != nil {
+			gatewayType, gatewayTypeOk := gatewayInner.Gateway.GetTypeOk()
+			if gatewayTypeOk && *gatewayType == (*management.ENUMGATEWAYTYPE_PING_FEDERATE.Ptr()) {
```

Comment:
It looks like this is getting a pointer to the enum and then immediately dereferencing it. Can we just use `management.ENUMGATEWAYTYPE_PING_FEDERATE` ?

Reply:
Fixed!